src/shiv/pip.py

```diff
@@ -3,9 +3,18 @@
 import subprocess
 import sys
 
+from pathlib import Path
 from typing import Generator, List
 
-from .constants import PIP_REQUIRE_VIRTUALENV, PIP_INSTALL_ERROR
+from .constants import PIP_REQUIRE_VIRTUALENV, PIP_INSTALL_ERROR, DISTUTILS_CFG_NO_PREFIX
+
+
+def pydistutils_cfg() -> Path:
+    """distutils can be configured by ``pydistutils.cfg`` or ``.pydistutils.cfg`` in a user's home directory."""
+    try:
+        return next(Path.home().glob("*pydistutils.cfg"))
+    except StopIteration:
+        return Path.home() / ".pydistutils.cfg"
```

Comment:
Don't we need to return either `pydistutils.cfg` OR `.pydistutils.cfg` depending on platform? 

distutils does it like so:

        if os.name == 'posix':
            user_filename = ".pydistutils.cfg"
        else:
            user_filename = "pydistutils.cfg"

I wouldn't mind a ternary, a la `return Path.home() / (".pydistutils.cfg" if os.name == "posix" else "pydistutils.cfg")`

Perhaps we can just skip this function alltogether and use this logic in the contextmanager, as I just realized there's no reason to search (glob) for the file anyway because we have to check if it exists either way.

(sorry for waffling so much!!)

Reply:
agreed!